### PR TITLE
Update isScalarTensor and krnl.load and krnl.store to hanlde both tensor<dtype> and tensor<1xdtype>

### LINKED ISSUE #2885: Compilation failing for static quantized Resnet50 Model on NNPA
**Problem**

Trying to compile a static quantized version of the [resentv2_50_Opset18.onnx] (https://github.com/onnx/models/tree/main/Computer_Vision/resnetv2_50_Opset18_timm) model fails with the below error:

> zdlc: /dlc-automation/onnx-mlir/src/Conversion/ONNXToKrnl/Math/Elementwise.cpp:2154: onnx_mlir::ONNXElementwiseUnaryOpLowering<mlir::ONNXDequantizeLinearOp>::matchAndRewrite(mlir::ONNXDequantizeLinearOp, onnx_mlir::ONNXElementwiseUnaryOpLowering<mlir::ONNXDequantizeLinearOp>::OpAdaptor, mlir::ConversionPatternRewriter&) const::<lambda(onnx_mlir::KrnlBuilder&, mlir::ValueRange)>: Assertion `isScalarValue(operands[i]) && "unary expected scalar additional values"' failed.

**Reproduce**
Attached is the [Sample Program](https://github.com/user-attachments/files/16355904/Archive.zip) for generating the static quantized model. 
Running the main.py
`python main.py --input_model resnetv2_50_Opset18.onnx --output_model resnetv2_50_Opset18_quant_output.onnx  --calibration_dataset ./test_images --preprocessed-input-model resnetv2_50_Opset18_pre.onnx`

The program generates a static quantized model with the name `resnetv2_50_Opset18_quant_output.onnx`.

On trying to compile the static quantized model on NNPA with the flags `--O3 --EmitLib --mtriple=s390x-ibm-loz --mcpu=z16 --maccel=NNPA` it fails with the above error.

### PR DESCRIPTION
**Pull request by tungld**
Scalar tensor can be tensor<dtype> or tensor<1xdtype>. This patch updates function isScalarTensor and builders for krnl.load and krnl.store to handle both cases.

Resolves #2885

### CODE REVIEW

**Comment by tungld** [comment]
@chentong319 with this patch, a krnl.iterate is not generated for `tensor<1xf32>` during onnx-to-krnl, so instead of generating
```mlir
krnl.iterate (%i = 0 to 1) {
   %v = krnl.load %arg[%i] : tensor<1xf32>
   krnl.store %v, %arg[%i] : tensor<1xf32> 
}
```
onnx-to-krnl pass generates
```mlir
   %v = krnl.load %arg[0] : tensor<1xf32>
   krnl.store %v, %arg[0] : tensor<1xf32>
```

and fusion does not work. I got this error `onnx-mlir-opt: /workdir/llvm-project/llvm/include/llvm/ADT/SmallVector.h:304: llvm::SmallVectorTemplateCommon::reference llvm::SmallVectorTemplateCommon<onnx_mlir::IndexExpr>::operator[](llvm::SmallVectorTemplateCommon::size_type) [T = onnx_mlir::IndexExpr]: Assertion `idx < size()' failed`

but it works well if I used `--disable-krnl-op-fusion`.

Do you have any suggestion to fix this error? e.g. does not do fusion for `tensor<1xdtype>`?

File: src/Dialect/Krnl/DialectBuilder.cpp [hunk 1]
```diff
@@ -56,6 +56,16 @@ static StringRef getFormat(const Type &inputType) {
 //====---------------- Support for Krnl Builder ----------------------===//
 
 Value KrnlBuilder::load(Value memref, ValueRange indices) const {
+  if (indices.size() == 0) {
+    // case memref<1xdtype>
+    MemRefType type = dyn_cast_or_null<MemRefType>(memref.getType());
+    assert(type && "Not MemRefType");
+    if (type.getRank() == 1 && type.getShape()[0] == 1) {
+      Value zero = b().create<arith::ConstantOp>(
```
Comment (AlexandreEichenberger) [inline src/Dialect/Krnl/DialectBuilder.cpp:64]: Nit: Can use the builder version
```
Value iZero = create.math.constantIndex(0);
```

**Review by AlexandreEichenberger — APPROVED** [review]
LGTM upon passing of the CI tests. Provided a "nit" for a more concise & readable code.

File: src/Dialect/Krnl/DialectBuilder.cpp (hunk 1 above)
Comment (tungld) [reply src/Dialect/Krnl/DialectBuilder.cpp]: Thanks! Done.

**Pull request merged**